### ISSUE TITLE
ES not using jvm.options file on custom config directory

### ISSUE THREAD

**Issue by juanfmx2** [elastic/elasticsearch#25964]
**Elasticsearch version:** 5.5.1
**JVM version:** jdk1.8.0_131
**OS version:** x86_64 GNU/Linux

I had a few elastic nodes (5.0.0) deployed on virtual machines that share an NFS. The virtual machines were sharing the same $JAVA_HOME and $ES_HOME installations on the NFS. I had a custom configuration for each node, and I was starting the ES nodes with a command like:
`$ES_HOME/bin/elasticsearch -Epath.conf=/my/super/secret/config`

I had set them up to use 16g of memory on their custom config directories. However, when I decided to use ES 5.5.1 I was having lots of java out of memory exceptions. I double checked the config files, and everything seem ok, until I finally noticed the nodes were starting with only 2g. Then, I updated the file under `$ES_HOME/config/jvm.options` and it took the changes.

In other words, I was expecting ES to load the **jvm.options** under `/my/super/secret/config` but it was using the one at `$ES_HOME/config`

**Comment by jasontedor** [comment]
To specify a custom jvm.options you must set the environment variable `ES_JVM_OPTIONS`. Starting in 6.0.0 this will no longer be necessary and Elasticsearch will do what you expect.
